src/components/RevokeContainer/RevokeContainer.tsx

```diff
@@ -0,0 +1,125 @@
+import { FunctionComponent, useState, useEffect } from "react";
+import { Redirect } from "react-router";
+import { useConfigContext } from "../../common/context/config";
+import { RevokeDocumentDropZone } from "./RevokeDocumentDropZone/RevokeDocumentDropZone";
+import { RevokeDocumentTileArea } from "./RevokeDocumentTileArea/RevokeDocumentTileArea";
+import { utils } from "@govtechsg/open-attestation";
+import { RevokeConfirmationModal } from "./RevokeConfirmationModal";
+import { RevokedScreen } from "./RevokedScreen/RevokedScreen";
+import { verificationBuilder, isValid, VerificationFragment, openAttestationVerifiers } from "@govtechsg/oa-verify";
+import { DocumentUploadState } from "../../constants/DocumentUploadState";
+
+const getNotRevokeFragment = (fragments: VerificationFragment[]): VerificationFragment[] =>
+  fragments.filter(
+    (status) => status.name !== "OpenAttestationEthereumDocumentStoreRevoked" && status.status !== "SKIPPED"
+  );
+
+export const RevokeContainer: FunctionComponent = () => {
+  const { config, setConfig } = useConfigContext();
+  const [revokeDocuments, setRevokeDocuments] = useState([]);
+  const [fileName, setFileName] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [showConfirmationModal, setShowConfirmationModal] = useState(false);
+  const [revokeStep, setRevokeStep] = useState(1);
+  const [documentUploadState, setDocumentUploadState] = useState(DocumentUploadState.INITIALIZED);
+
+  useEffect(() => {
+    if (revokeDocuments.length > 0) {
+      const isDocumentRevokable = utils.isDocumentRevokable(revokeDocuments[0]);
+
+      const validateDocument = async () => {
+        const network = config?.network || "homestead";
+        if (network !== "local") {
+          const verify = verificationBuilder(openAttestationVerifiers, { network: network });
+          const fragments = await verify(revokeDocuments[0]);
+          const notRevokeFragments = getNotRevokeFragment(fragments);
+          const hashValid = isValid(fragments, ["DOCUMENT_INTEGRITY"]);
+          const issuedValid = isValid(notRevokeFragments, ["DOCUMENT_STATUS"]);
+          const identityValid = isValid(fragments, ["ISSUER_IDENTITY"]);
+          const isDocumentValid = hashValid && issuedValid && identityValid;
+
+          if (!isDocumentValid) {
+            setErrorMessage("Error: Document is not valid, please use a valid document.");
+            setRevokeDocuments([]);
+            setDocumentUploadState(DocumentUploadState.ERROR);
+          } else {
+            setErrorMessage("");
+            setDocumentUploadState(DocumentUploadState.DONE);
+          }
+        } else {
+          setErrorMessage("");
+          setDocumentUploadState(DocumentUploadState.DONE);
```

Comment:
for simplicity, don't have to control error messages here. leave it to the error component at `RevokeDocumentDropZone.tsx`